app/cdap/components/hydrator/components/LeftPanel/LeftPanel.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useState } from 'react';
+import ChevronLeft from '@material-ui/icons/ChevronLeft';
+import ChevronRight from '@material-ui/icons/ChevronRight';
+import Select from '@material-ui/core/Select';
+import SidePanel from 'components/hydrator/components/SidePanel/SidePanel';
+import MenuItem from '@material-ui/core/MenuItem';
+import { Button } from '@material-ui/core';
+import styled from 'styled-components';
+
+interface ILeftPanelProps {
+  onArtifactChange: (value: any) => void;
+  pluginsMap: any[];
+  selectedArtifact: any;
```

Comment:
Any chance to add types for these?

Reply:
I don't have a good understanding of the artifact types - would be better suited to do this up the line a bit (like where the data is requested).